### PR TITLE
add placeholder to RichChoiceView

### PR DESCRIPTION
**Pull request by hchoura**

### CODE REVIEW

**Comment by Nauna** [comment]
Instead of adding a placeholder property on RichChoiceView what do you think about reusing the choosePlaceholder property. It's a more consistent msg then having the dashes as default:

image

But otherwise, nicely done. Code functions well.

File: src/foam/u2/view/RichChoiceView.js [hunk 1]
```diff
@@ -298,7 +298,7 @@ foam.CLASS({
       documentation: 'Replaces choose from placeholder with passed in string.',
       expression: function(of) {
         var plural = of.model_.plural.toLowerCase();
-        return this.CHOOSE_FROM + plural + '...';
+        return this.CHOOSE_FROM + plural + ' ...';
```
Comment (TW80000) [inline src/foam/u2/view/RichChoiceView.js:301]: I disagree with this change.

File: src/foam/u2/view/RichChoiceView.js [hunk 2]
```diff
@@ -343,13 +349,15 @@ foam.CLASS({
...
                     self.isOpen_ = ! self.isOpen_;
+                    this.placeholder = self.placeholder;
                   }
                 })
                 .start()
                   .addClass(this.myClass('custom-selection-view'))
-                  .add(this.slot((data) => {
-                    return this.E().tag(self.selectionView, {
+                  .add(this.slot((data, isOpen_) => {
+                   return this.E().tag(self.selectionView, {
```
Comment (TW80000) [inline src/foam/u2/view/RichChoiceView.js]:
```suggestion
                    return this.E().tag(self.selectionView, {
```

File: src/foam/u2/view/RichChoiceView.js [hunk 3]
```diff
@@ -343,13 +349,15 @@ foam.CLASS({
...
                       data: data,
+                      defaultSelectionPrompt: self.placeholder,
```
Comment (TW80000) [inline src/foam/u2/view/RichChoiceView.js]: We're already binding `defaultSelectionPrompt` two lines below.

File: src/foam/u2/view/RichChoiceView.js [hunk 4]
```diff
@@ -392,7 +400,19 @@ foam.CLASS({
                         .start().hide(!! section.hideIfEmpty && resp[index].value <= 0 || ! section.heading)
                           .addClass(self.myClass('heading'))
                           .add(section.heading)
-                        .end()
+                          .end()
```
Comment (TW80000) [inline src/foam/u2/view/RichChoiceView.js]: Fix indentation. This and the code you added below should be indented one level less than it is currently.

File: src/foam/u2/view/RichChoiceView.js [hunk 5]
```diff
@@ -343,13 +349,15 @@ foam.CLASS({
                 .on('click', function() {
                   if ( self.mode === foam.u2.DisplayMode.RW ) {
                     self.isOpen_ = ! self.isOpen_;
+                    this.placeholder = self.placeholder;
```
Comment (TW80000) [inline src/foam/u2/view/RichChoiceView.js]: What does this do?

File: src/foam/u2/view/RichChoiceView.js [hunk 6]
```diff
@@ -392,7 +400,19 @@ foam.CLASS({
...
+                          .end()
+                          .start()
+                            .addClass(self.myClass('selection-view'))
+                            .add('--------------------------')
+                                .callIf(! section.disabled, function() {
+                                  this.on('click', () => {
+                                    self.fullObject_ = undefined;
+                                    self.data = 0;
+                                    self.isOpen_ = false;
+                                    self.cross=true;
```
Comment (TW80000) [inline src/foam/u2/view/RichChoiceView.js]: Please remove.

**Review by TW80000 — CHANGES_REQUESTED** [review]
Rather than add an option to the first section in the dropdown, I think we should either add a button in the dropdown that shows above the first section or include a button to clear in the button part itself.

File: src/foam/u2/view/RichChoiceView.js (hunk 1 above)
Comment (hchoura) [reply src/foam/u2/view/RichChoiceView.js:301]: we need to add space between word and Ellipsis.

File: src/foam/u2/view/RichChoiceView.js (hunk 6 above)
Comment (hchoura) [reply src/foam/u2/view/RichChoiceView.js]: we need to close the close the option list

**Comment by TW80000** [comment]
Closing since #3090 was merged.

**Pull request closed without merging**